crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -160,42 +203,81 @@ pub(crate) fn unused_import(checker: &Checker, scope: &Scope, diagnostics: &mut
     }
 
     let in_init = checker.path().ends_with("__init__.py");
-    let fix_init = !checker.settings.ignore_init_module_imports;
+    let fix_init = checker.settings.preview.is_enabled();
+    // TODO: find the `__all__` node
+    let dunder_all = None;
 
-    // Generate a diagnostic for every import, but share a fix across all imports within the same
+    // Generate a diagnostic for every import, but share fixes across all imports within the same
     // statement (excluding those that are ignored).
-    for ((node_id, exceptions), imports) in unused {
+    for ((import_statement, exceptions), bindings) in unused {
         let in_except_handler =
             exceptions.intersects(Exceptions::MODULE_NOT_FOUND_ERROR | Exceptions::IMPORT_ERROR);
-        let multiple = imports.len() > 1;
+        let multiple = bindings.len() > 1;
 
-        let fix = if (!in_init || fix_init) && !in_except_handler {
-            fix_imports(checker, node_id, &imports, in_init).ok()
+        // pair each binding with context; divide them by how we want to fix them
+        let (to_reexport, to_remove): (Vec<_>, Vec<_>) = bindings
+            .into_iter()
+            .map(|binding| {
+                let context = if in_except_handler {
+                    Some(UnusedImportContext::ExceptHandler)
+                } else if in_init {
+                    Some(UnusedImportContext::Init {
+                        first_party: is_first_party(
+                            checker,
+                            &binding.import.qualified_name().to_string(),
```

Comment:
Just trying to understand what's happening in Latch, at least one thing is that I think `isort::categorize` expects `.module_name()` rather than `.qualified_name()`, so this should probably be `binding.import.module_name()` (although I wouldn't expect it to change the outcome -- the module name is generally a prefix of the qualified name, and the import categorization just operates on the first segment).